### ISSUE TITLE
ctypes: isspace recognizes some Unicode characters as white-space

### ISSUE THREAD

**Issue by NonStandardModel** [SerenityOS/serenity#7425]
The glyph for letter `Č (U+010C)` and `č (U+010D)` will not be rendered correctly until next printable character is rendered. Glyphs will remain corrupted if they are followed by white-space, tab or newline.
I have noted this issue in Text Editor and in Hack Studio.
For ease of testing: letter `č` on Slovenian keymap (`sl.json`) is where `ö` is on German or `;` on English.

![Issue_Glyph_1](https://user-images.githubusercontent.com/33322918/119273413-2b2d8e00-bc0b-11eb-817d-b197d2c47fae.jpg)
![Issue_Glyph_2](https://user-images.githubusercontent.com/33322918/119273427-37195000-bc0b-11eb-9026-942b21136d6b.jpg)

**Comment by NonStandardModel** [comment]
I have also noted that **IRC Client** will display `č/Č` as _white-space_. 
So the issue is deeper that just text rendering. I am not sure what the title of this issue should be. 
I have a feeling `č/Č` is recognized as _white-space_ by something and thus it is tinted red in examples in first comment. But I am not able to substantiate my claim. I have been also trying with other 'non-ascii' characters, but haven't yet found any that would have similar issue.
![Issue_Glyph_3](https://user-images.githubusercontent.com/33322918/120081454-12ffb800-c0be-11eb-879d-15db513ff930.jpg)

**Comment by NonStandardModel** [comment]
After a bit of debug I think I have found the core of this issue. The core reason that some characters are recognized as white-space comes from cast to unsigned char here (ctypes.h):
```
static inline int __inline_isspace(int c)
{
    return _ctype_[(unsigned char)(c)] & (_S);
}
```
I have found few other characters that should have the same problem, but I do not know if all are used in any existing keymaps. The "offending" characters are `U+0109, U+010A, U+010B, U+010C, U+010D`. Due to the cast this will be "seen" by `__inline_isspace` as 0x9, ..., 0xD. I suspect that this are not the only ones.

Anyhow, in today's YouTube video of new Spec-Compliant URL parser I have seen that different `isspace` function is proposed, so it may solve this issue.

**Comment by NonStandardModel** [comment]
The issues inhere will be probably solved with https://github.com/SerenityOS/serenity/pull/7684 .

**Comment by MaxWipfli** [comment]
> The issues inhere will be probably solved with #7684 .

Yes, they should be solved. Although there is another issue, #7315, which is also responsible for some glyphs to be shown as whitespace instead of `?` (replacement character). Since those only apply to the larger-size fonts, it should not be the problem here.

**Comment by NonStandardModel** [comment]
https://github.com/SerenityOS/serenity/pull/7684 was merged and it solved the issues. This can be closed.
